core/app/models/workarea/bulk_action/product_edit.rb

```diff
@@ -52,13 +58,27 @@ def apply_pricing!(product)
 
         pricing_skus(product).each do |sku|
           sku_pricing = pricing.dup
-          prices = sku_pricing.delete('prices') unless sku.prices.blank?
+          prices = sku_pricing.delete('prices')
           sku.update_attributes!(sku_pricing)
 
-          if prices.present?
-            sku.prices.select(&:generic?).each do |price|
-              price.update_attributes!(prices.first)
+          if prices.present? && sku.prices.present?
+            sku.prices.each do |price|
+              updates = prices.each_with_object({}) do |(type, value), params|
```

Comment:
This code feels like there's a missing class/object due to all the methods written into this class to support it. Try to get it to something like this maybe:
```
sku.prices.each do |price|
  update = PriceUpdate.new(price, prices)
  price.update!(update.attributes) if update.apply?
end
```

Reply:
yeah I was thinking about adding a new class to encapsulate all this logic.